birdhouse/optional-components/testthredds/default.env

```diff
@@ -25,3 +25,9 @@ OPTIONAL_VARS="
   \$TESTTHREDDS_PORT
   \$TESTTHREDDS_NAME
 "
+
+# add any component that this component requires to run
+COMPONENT_DEPENDENCIES="
+  $COMPONENT_DEPENDENCIES
+  ./config/thredds
```

Comment:
Not sure why testthredds would require thredds.  They are 2 separate services.  But I have a feeling you might be right, I just can not think of a reason right now.

Reply:
It's using the config files under `config/thredds`. I guess it's not strictly necessary for the actual thredds service to be running but it does need to include those config files so that the right environment variables get added in. 

What if we symlinked the relevant files from config/thredds into optional-components/testthredds instead?

Reply:
If only configs files are required, no symlink needed either, `testthredds` can simply volume-mount the config files path from `thredds` since they are in the same repo anyways.  `thredds` will always be present to share its configs with `testthredds`.  I think the template expansion works on all components, regardless if they are enabled for now.

```
volumes:
  - ./optional-components/testthredds/some.cfg:/some/path
  - ./config/thredds/some.cfg:/some/path2
  - (...)
```
This might look weird to mix match `testthredds` and `thredds` config but since they are in the same repo, those paths will always resolve so I think it's probably safe to do this.  But a warning comments would be good.

Reply:
At the moment, template expansion only occurs if the directory is in the ALL_CONF_DIRS list so it is possible to configure testthredds to run without expanding the template files in config/thredds/.

I can add some additional logic so that the template files in thredds are expanded if testthredds is enabled. But I'm wondering if the reason that testthredds uses the same config files is because we actually want them to use the exact same settings or if it's simply easier to not have to include the files twice?

If it's the latter, then maybe the better way is to actually copy those files to testthredds. Then if we ever want to configure thredds and testthredds differently, one isn't tied to the other.

Reply:
The reason `testthredds` use the same config as `thredds` is simply my laziness in duplicating the configs and to avoid having to repeat the same change twice if ever we need to change Thredds related configs.

It does not have to be this way.  If you think duplicating the config files to properly make them independent is better, I have no objections.